internal/venice-test-common/src/integrationtest/java/com/linkedin/venice/endToEnd/WriteComputeWithActiveActiveReplicationTest.java

```diff
@@ -0,0 +1,1184 @@
+    sendStreamingRecord(systemProducerMap.get(childDatacenters.get(0)), storeName, "marker1", val2);
+    verifyFLMRecord(storeName, dc1RouterUrl, "marker1", "", Collections.emptyList(), Collections.emptyMap());
+    verifyFLMRecord(storeName, dc0RouterUrl, "marker1", "", Collections.emptyList(), Collections.emptyMap());
+    verifyFLMRecord(storeName, dc1RouterUrl, key2, regularFieldValue, arrayFieldValue, expectedMapFieldValue);
+    verifyFLMRecord(storeName, dc0RouterUrl, key2, regularFieldValue, arrayFieldValue, expectedMapFieldValue);
+
+    // PartialPut: When PartialPut with the same timestamp as PUT
+    // todo: Currently this wipes out the data with the timestamp <= t2 including Puts timestamp. This
+    // creates a race between PartialPut and PUT
+    /*
+    ub = new UpdateBuilderImpl(wcSchemaV1);
+    mapFieldValue = new HashMap<>();
+    mapFieldValue.put("six", 6);
+    mapFieldValue.put("seven", 7);
+    ub.setNewFieldValue(MAP_FIELD, mapFieldValue);
+    timestampedOp = new VeniceObjectWithTimestamp(ub.build(), 2);
+    sendStreamingRecord(systemProducerMap.get(childDatacenters.get(0)), storeName, key2, timestampedOp);
+    // send another record from the same region/producer to ensure that the previous operation was completed
+    sendStreamingRecord(systemProducerMap.get(childDatacenters.get(0)), storeName, "marker1", val2);
+    verifyFLMRecord(storeName, dc1RouterUrl, "marker1", "", Collections.emptyList(), Collections.emptyMap());
+    verifyFLMRecord(storeName, dc0RouterUrl, "marker1", "", Collections.emptyList(), Collections.emptyMap());
+    verifyFLMRecord(storeName, dc1RouterUrl, key2, regularFieldValue, arrayFieldValue, expectedMapFieldValue);
+    verifyFLMRecord(storeName, dc0RouterUrl, key2, regularFieldValue, arrayFieldValue, expectedMapFieldValue);
+    */
+
+    // PartialPut: elements with <= to the timestamp of PartialPut will be removed and elements from the PartialPut
+    // will be added
+    ub = new UpdateBuilderImpl(wcSchemaV1);
+    mapFieldValue = new HashMap<>();
+    mapFieldValue.put("six", 6);
+    mapFieldValue.put("seven", 7);
+    ub.setNewFieldValue(MAP_FIELD, mapFieldValue);
+    timestampedOp = new VeniceObjectWithTimestamp(ub.build(), 3);
+    sendStreamingRecord(systemProducerMap.get(childDatacenters.get(1)), storeName, key2, timestampedOp);
+    expectedMapFieldValue.putAll(mapFieldValue);
+    expectedMapFieldValue.remove("two"); // existing timestamp 2
+    expectedMapFieldValue.remove("three"); // existing timestamp 2
+    expectedMapFieldValue.remove("ThreeZero"); // existing timestamp 3
+    verifyFLMRecord(storeName, dc1RouterUrl, key2, regularFieldValue, arrayFieldValue, expectedMapFieldValue);
+    verifyFLMRecord(storeName, dc0RouterUrl, key2, regularFieldValue, arrayFieldValue, expectedMapFieldValue);
+
+    // PartialPut
+    // todo: Yet another race between two partial updates
+    /*
```

Comment:
Is this resolved?

Reply:
No this is one of the race conditions. I'm still looking at it. Maybe I'll create a ticket for it.